### PR TITLE
Authorize users to to be added and access organizations

### LINKED ISSUE #105: OrganizationAuthorizedConstraint doesn't work
`OrganizationAuthorizedConstraint` searches the URL path for the first set of numeric characters and tries to look up an org with an id matching these digits. If no organization is found the check fails open. The `OrganizationsController#set_organization` method uses `Organization.find(params[:id])` to lookup organizations. Because of FriendlyId, this can take a slug instead of a numeric id. The result is that authorization is not checked for URLs using the org slug instead of a numeric ID.

The authorization check should be moved to the controller level and the organization should be looked up in the same way as elsewhere in the controller. The check should also fail closed if the organization can't be found.

### PR DESCRIPTION
**Pull request by tarebyte**
Closes #105 

This is the implementation that was discussed in #107.

When a user goes to an organization page on classroom if they are not already apart of it, Classroom will try to verify that they are in fact an organization admin. If they are it will add them to the org, then take them to the org page.

This is where I get unsure, right now when a user goes to an Assignment or GroupAssignment page if they aren't part of the org it will just 404. However this can be easily changed so that it will add the new user to the org if they go to any of the pages dealing with the org.

Thoughts?

/cc @johndbritton @mastahyeti

### CODE REVIEW

**Comment by johndbritton** [comment]
> However this can be easily changed so that it will add the new user to the org if they go to any of the pages dealing with the org.

I don't see a problem with checking auth for any url under `/organization/ORG` and automatically adding the user if they have they are admin on the org.

File: app/controllers/organization_authorized_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class OrganizationAuthorizedController < ApplicationController
+  before_action :set_organization, :authorize_organization_access
+
+  decorates_assigned :organization
+
+  private
+
+  def authorize_organization_access
+    not_found if !@organization.users.include?(current_user) && current_user.staff?
```
Comment (mastahyeti) [inline app/controllers/organization_authorized_controller.rb]: This looks like it only checks authorization if the user is staff. Maybe

```ruby
not_found unless @organization.users.include?(current_user) || current_user.staff?
```

Comment (johndbritton) [inline app/controllers/organization_authorized_controller.rb]: @mastahyeti beat me to it, I had a comment box open for that line and was trying to parse it mentally.

Comment (tarebyte) [inline app/controllers/organization_authorized_controller.rb]: @mastahyeti @johndbritton this logic went away with my latest commit

File: app/controllers/assignment_invitations_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class AssignmentInvitationsController < InvitationsController
+  skip_before_action :set_organization, :authorize_organization_access
```
Comment (johndbritton) [inline app/controllers/assignment_invitations_controller.rb]: :+1: Good move switching to an opt-out here.

Comment (tarebyte) [inline app/controllers/assignment_invitations_controller.rb]: This is actually leftover code from another idea I was spiking out. `set_organization` never actually happens in this controller.

Comment (johndbritton) [inline app/controllers/assignment_invitations_controller.rb]: Never mind me then.

**Comment by tarebyte** [comment]
@johndbritton this is what the invite page for the organization looks like now. Any feedback on the copy would be much appreciated.

screen shot 2015-08-27 at 11 04 26

**Comment by johndbritton** [comment]
Looks good. The copy could use some information about why someone would want to add another admin to classroom. Maybe explain a bit that admins are able to create assignments and get lists of submissions by students.

**Comment by johndbritton** [comment]
If this is going to be an interstitial page in adding an org, I think the "Skip for now" button should be moved down to the bottom of the page to fit a more normal UX pattern.

cc/ @jasonlong

**Comment by tarebyte** [comment]
@johndbritton something like this?

screen shot 2015-08-27 at 13 04 25

**Comment by tarebyte** [comment]
screen shot 2015-08-27 at 13 10 56

**Comment by johndbritton** [comment]
:+1: Works for me, we might want to revise at some point in the future but this is good.

**Pull request merged**